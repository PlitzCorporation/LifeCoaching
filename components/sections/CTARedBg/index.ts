import CTARedBg from './CTARedBg';

export default CTARedBg;
