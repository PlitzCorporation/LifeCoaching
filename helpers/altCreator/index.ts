const altCreator = (imageUri: string) => {
	const altText = imageUri
		.replace('uploads/Reason4HopeCoaching/', '')
		.replace(/\.(webp|jpg|png|svg)/g, '')
		.replace(/-/g, ' ');

	return altText;
};

export default altCreator;
