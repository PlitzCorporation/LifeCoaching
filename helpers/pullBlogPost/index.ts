export const getAllBlogPosts = async () => {
	const res = await fetch(`${process.env.NEXT_PUBLIC_URL}/td/api/blogposts`, {
		next: { revalidate: 60 },
		headers: { storedId: 'Reason4HopeCoaching' },
	});

	const data = await res.json();

	return data;
};

export const getPostByLink = async (link: string) => {
	const res = await fetch(
		`${process.env.NEXT_PUBLIC_URL}/td/api/blogposts/${link}`,
		{
			next: { revalidate: 60 },
			headers: { storedId: 'Reason4HopeCoaching' },
		},
	);

	if (res.status === 404) {
		return new Error('Page not found');
	}

	if (!res.ok) {
		return new Error('An error occurred while fetching the data');
	}

	const data = await res.json();

	return data;
};

export const getPostsByCategory = async (categoryLink: string) => {
	const res = await fetch(
		`${process.env.NEXT_PUBLIC_URL}/td/api/categories/${categoryLink}`,
		{
			next: { revalidate: 60 },
			headers: { storedId: 'Reason4HopeCoaching' },
		},
	);

	const data = await res.json();

	return data;
};
