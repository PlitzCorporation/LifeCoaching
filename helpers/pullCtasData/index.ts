export const getAllCtas = async () => {
	const res = await fetch(`${process.env.NEXT_PUBLIC_CMS}/api/sliders`, {
		next: { revalidate: 60 },
		headers: { storedId: 'Reason4HopeCoaching' },
	});

	const data = await res.json();

	return data;
};
