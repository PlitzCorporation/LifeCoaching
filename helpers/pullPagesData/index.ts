export const getAllPages = async () => {
	const res = await fetch(`${process.env.NEXT_PUBLIC_URL}/td/api/pages`, {
		next: { revalidate: 60 },
		headers: { storedId: 'Reason4HopeCoaching' },
	});

	const data = await res.json();

	return data;
};

export const getPageByLink = async (link: string) => {
	const res = await fetch(
		`${process.env.NEXT_PUBLIC_URL}/td/api/pages/${link}`,
		{
			next: { revalidate: 60 },
			headers: { storedId: 'Reason4HopeCoaching' },
		},
	);

	if (res.status === 404) {
		return new Error('Page not found');
	}

	if (!res.ok) {
		return new Error('An error occurred while fetching the data');
	}

	const data = await res.json();

	return data;
};
