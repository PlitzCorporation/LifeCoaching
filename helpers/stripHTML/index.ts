const stripHTML = (content: string) => {
	const plainContent = content?.replace(/<\/*[^>]+(>|$)/g, '') ?? '';

	return plainContent;
};

export default stripHTML;
