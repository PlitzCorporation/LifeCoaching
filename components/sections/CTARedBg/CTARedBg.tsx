import React from 'react';

import { getAllCtas } from '@/helpers/pullCtasData';

import OutlinedButton from '@/components/ui/OutlinedButton';
import SolidButton from '@/components/ui/SolidButton';

import { CTAProps } from '@/types/ctas';

const CTARedBg = async () => {
	const cta: CTAProps[] = await getAllCtas();

	const ctaData = cta.filter((cta) => cta.order === 1)[0];

	return (
		<section className="bg-gradient-to-b from-plitz-primary to-[#4B1322] py-16 lg:py-32">
			<div className="plitz-container flex flex-col items-center text-center">
				<h3 className="text-plitz-secondary max-w-4xl mb-5">{ctaData.title}</h3>
				<div
					dangerouslySetInnerHTML={{ __html: ctaData.details }}
					className="text-white text-balance drop-shadow-plitz-darker max-w-4xl mb-10 lg:mb-16"
				></div>
				<div className="flex justify-center items-center">
					{ctaData.buttons
						.sort((a, b) => a.position - b.position)
						.map((button) =>
							button.primary ? (
								<SolidButton
									key={button.link}
									href={button.link}
									label={button.label}
									extraClasses="bg-white [&>span]:text-plitz-primary [&>span]:hover:text-plitz-accent"
								/>
							) : (
								<OutlinedButton
									key={button.link}
									href={button.link}
									label={button.label}
									extraClasses="border-white [&>span]:text-white"
								/>
							),
						)}
				</div>
			</div>
		</section>
	);
};

export default CTARedBg;
