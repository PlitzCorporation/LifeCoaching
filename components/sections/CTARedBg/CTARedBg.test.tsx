import React from 'react';
import { render } from '@testing-library/react';

import { getAllCtas } from '@/helpers/pullCtasData';
import CTARedBg from './CTARedBg';

// Mock the getAllCtas function
jest.mock('@/helpers/pullCtasData', () => ({
	getAllCtas: jest.fn().mockImplementation(() =>
		Promise.resolve([
			{
				order: 1,
				title: 'CTA Title',
				details: '<p>CTA Details</p>',
				buttons: [
					{ link: '/link1', label: 'Button 1', primary: true, position: 1 },
					{ link: '/link2', label: 'Button 2', primary: false, position: 2 },
				],
			},
		]),
	),
}));

describe('CTARedBg Component', () => {
	it('renders correctly with CTA data', async () => {
		// Arrange
		const mockCtas = [
			{
				order: 1,
				title: 'CTA Title',
				details: '<p>CTA Details</p>',
				buttons: [
					{ link: '/link1', label: 'Button 1', primary: true, position: 1 },
					{ link: '/link2', label: 'Button 2', primary: false, position: 2 },
				],
			},
		];
		(getAllCtas as jest.Mock).mockResolvedValue(mockCtas);

		// Act
		const ComponentResolved = await CTARedBg();

		const MockComponent = () => ComponentResolved;
		const { container } = render(<MockComponent />);

		expect(container).toMatchSnapshot();
	});
});
