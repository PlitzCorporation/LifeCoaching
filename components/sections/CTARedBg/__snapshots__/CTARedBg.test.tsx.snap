// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTARedBg Component renders correctly with CTA data 1`] = `
<div>
  <section
    class="bg-gradient-to-b from-plitz-primary to-[#4B1322] py-16 lg:py-32"
  >
    <div
      class="plitz-container flex flex-col items-center text-center"
    >
      <h3
        class="text-plitz-secondary max-w-4xl mb-5"
      >
        CTA Title
      </h3>
      <div
        class="text-white text-balance drop-shadow-plitz-darker max-w-4xl mb-10 lg:mb-16"
      >
        <p>
          CTA Details
        </p>
      </div>
      <div
        class="flex justify-center items-center"
      >
        <a
          class="relative flex items-center bg-plitz-primary text-white px-5 py-3 hover:rounded-xl border border-plitz-primary md:px-7 lg:px-9 md:py-4 hover:border-plitz-accent hover:text-plitz-accent hover:bg-white transition-all hover:transition-all group shadow-2xl bg-white [&>span]:text-plitz-primary [&>span]:hover:text-plitz-accent"
          data-testid="aTag"
          href="/link1"
        >
          <span
            class="translate-x-2 group-hover:-translate-x-1 transition-all"
          >
            Button 1
          </span>
          <svg
            class="fill-white translate-x-0 opacity-0 group-hover:fill-plitz-accent group-hover:opacity-100 group-hover:translate-x-1 transition-all"
            height="1em"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M440.6 273.4c4.7-4.5 7.4-10.8 7.4-17.4s-2.7-12.8-7.4-17.4l-176-168c-9.6-9.2-24.8-8.8-33.9.8s-8.8 24.8.8 33.9L364.1 232H24c-13.3 0-24 10.7-24 24s10.7 24 24 24h340.1L231.4 406.6c-9.6 9.2-9.9 24.3-.8 33.9s24.3 9.9 33.9.8l176-168z"
            />
          </svg>
        </a>
        <a
          class="relative flex items-center bg-transparent text-white px-9 py-3 lg:py-4 hover:rounded-xl border border-plitz-primary hover:border-plitz-accent transition-all group shadow-2xl border-white [&>span]:text-white"
          data-testid="aTag"
          href="/link2"
        >
          <span
            class="translate-x-2 group-hover:-translate-x-1 transition-all text-plitz-primary group-hover:text-plitz-accent"
          >
            Button 2
          </span>
          <svg
            class="fill-plitz-primary translate-x-0 opacity-0 group-hover:fill-plitz-accent group-hover:opacity-100 group-hover:translate-x-1 transition-all"
            height="1em"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M440.6 273.4c4.7-4.5 7.4-10.8 7.4-17.4s-2.7-12.8-7.4-17.4l-176-168c-9.6-9.2-24.8-8.8-33.9.8s-8.8 24.8.8 33.9L364.1 232H24c-13.3 0-24 10.7-24 24s10.7 24 24 24h340.1L231.4 406.6c-9.6 9.2-9.9 24.3-.8 33.9s24.3 9.9 33.9.8l176-168z"
            />
          </svg>
        </a>
      </div>
    </div>
  </section>
</div>
`;
